app/services/foreman_salt/fact_parser.rb

```diff
@@ -84,6 +84,10 @@ def support_interfaces_parsing?
       true
     end
 
+    def boot_timestamp
+      Time.zone.now.to_i - facts[:uptime_seconds].to_i
```

Comment:
Unfortunately, I don't think this will work because the facts (= grains in saltstack) don't include the uptime. To receive the uptime, 'status.uptime' needs to be called, like 

salt 'myawesomehost.dev.atix' status.uptime

See https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.status.html#salt.modules.status.uptime

This needs to be done on the smart_proxy_salt side and maybe merged with the general facts. Or another smart_proxy API needs to be introduced to receive the uptime.

Reply:
oh, I was under the impression there was this fact present for hosts, I'll close this PR then, my salt knowledge is very limited and I'm afraid I won't be able to path smart_proxy_salt